File: backend/app/schemas/medication_catalog.py
"""Medication catalog document schema (drug list for compatibility/UI)."""
from datetime import datetime, timezone
from typing import Optional

from pydantic import BaseModel, Field

from .objectid import PyObjectId


def _utc_now() -> datetime:
    return datetime.now(timezone.utc)


class MedicationCatalogDoc(BaseModel):
    """Mongo document shape for medications (catalog) collection."""

    id: Optional[PyObjectId] = Field(None, alias="_id")
    medication_id: Optional[str] = None
    name: str
    brand_name: Optional[str] = None
    description: Optional[str] = None
    side_effects: list[str] = Field(default_factory=list)
    warnings: list[str] = Field(default_factory=list)
    how_to_use: Optional[str] = None
    created_at: datetime = Field(default_factory=_utc_now)
    updated_at: datetime = Field(default_factory=_utc_now)

    model_config = {"populate_by_name": True, "arbitrary_types_allowed": True}

File: express-backend/src/models/MedicationCatalog.ts
import mongoose, { Schema, type Document, type Model } from 'mongoose';

export interface MedicationCatalogDocument extends Document {
  medication_id?: string;
  name: string;
  brand_name?: string;
  description?: string;
  side_effects: string[];
  warnings: string[];
  how_to_use?: string;
  created_at: Date;
  updated_at: Date;
}

const MedicationCatalogSchema = new Schema<MedicationCatalogDocument>(
  {
    medication_id: { type: String, index: true },
    name: { type: String, required: true },
    brand_name: { type: String },
    description: { type: String },
    side_effects: { type: [String], default: [] },
    warnings: { type: [String], default: [] },
    how_to_use: { type: String },
    created_at: { type: Date, default: () => new Date() },
    updated_at: { type: Date, default: () => new Date() },
  },
  {
    collection: 'medications',
  },
);

MedicationCatalogSchema.pre('save', function (next) {
  this.updated_at = new Date();
  next();
});

export const MedicationCatalog: Model<MedicationCatalogDocument> =
  mongoose.models.MedicationCatalog ||
  mongoose.model<MedicationCatalogDocument>('MedicationCatalog', MedicationCatalogSchema);

